cmd/operator/main.go

```diff
@@ -256,3 +280,101 @@ func main() {
+func addCertificateRotation(ctx context.Context, mgr manager.Manager, secretName, certDir, operatorServiceName, metricsServerServiceName, webhooksServiceName string) {
+	caName := "kedaorg-ca"
+	caOrganization := "kedaorg"
+
+	var rotatorHooks = []rotator.WebhookInfo{
+		{
+			Name: "keda-admission",
+			Type: rotator.Validating,
+		},
+	}
+
+	// Make sure certs are generated and valid if cert rotation is enabled.
+	setupFinished := make(chan struct{})
+	ensureSecret(ctx, mgr, secretName)
+	extraDNSNames := []string{}
+	extraDNSNames = append(extraDNSNames, getDNSNames(operatorServiceName)...)
+	extraDNSNames = append(extraDNSNames, getDNSNames(webhooksServiceName)...)
+	extraDNSNames = append(extraDNSNames, getDNSNames(metricsServerServiceName)...)
+
+	setupLog.V(1).Info("setting up cert rotation")
+	if err := rotator.AddRotator(mgr, &rotator.CertRotator{
+		SecretKey: types.NamespacedName{
+			Namespace: kedautil.GetPodNamespace(),
+			Name:      secretName,
+		},
+		CertDir:                certDir,
+		CAName:                 caName,
+		CAOrganization:         caOrganization,
+		DNSName:                extraDNSNames[0],
+		ExtraDNSNames:          extraDNSNames,
+		IsReady:                setupFinished,
+		Webhooks:               rotatorHooks,
+		RestartOnSecretRefresh: true,
+		RequireLeaderElection:  true,
+	}); err != nil {
+		setupLog.Error(err, "unable to set up cert rotation")
+		os.Exit(1)
+	}
+}
+
+// getDNSNames  creates all the possible DNS names for a given service
+func getDNSNames(service string) []string {
+	namespace := kedautil.GetPodNamespace()
+	return []string{
+		service,
+		fmt.Sprintf("%s.%s", service, namespace),
+		fmt.Sprintf("%s.%s.svc", service, namespace),
+		fmt.Sprintf("%s.%s.svc.local", service, namespace),
+	}
+}
+
+// ensureSecret ensures that the secret used for storing TLS certificates exists
+func ensureSecret(ctx context.Context, mgr manager.Manager, secretName string) {
+	secrets := &corev1.SecretList{}
+	kedaNamespace := kedautil.GetPodNamespace()
+	opt := &client.ListOptions{
+		Namespace: kedaNamespace,
+	}
+
+	err := mgr.GetAPIReader().List(ctx, secrets, opt)
+	if err != nil {
+		setupLog.Error(err, "unable to check secrets")
+		os.Exit(1)
+	}
+
+	exists := false
+	for _, secret := range secrets.Items {
+		if secret.Name == secretName {
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		secret := &corev1.Secret{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      secretName,
+				Namespace: kedaNamespace,
+				Labels: map[string]string{
+					"app":                         "keda-operator",
+					"app.kubernetes.io/name":      "keda-operator",
+					"app.kubernetes.io/component": "keda-operator",
+					"app.kubernetes.io/part-of":   "keda",
+				},
+			},
+		}
+		err = mgr.GetClient().Create(ctx, secret)
+		if err != nil {
+			setupLog.Error(err, "unable to create certificates secret")
+			os.Exit(1)
```

Comment:
return err and exit in the main

Reply:
nice catch! I moved this code and I forgot the exit part